tests/testthat/test-db-sql.R

```diff
@@ -19,3 +19,15 @@ test_that("sql_query_rows() works", {
     sql("SELECT COUNT(*) FROM `abc` AS `master`")
   )
 })
+
+test_that("handles DBI error", {
+  con <- DBI::dbConnect(RSQLite::SQLite(), ":memory:")
+  on.exit(DBI::dbDisconnect(con))
+
+  expect_error(db_analyze(con, "tbl"), class = "dbplyr_error_dbi")
```

Comment:
And I think these are probably better as snapshot tests as that helps you see whether or not the `call` has been correctly passed through.

Reply:
The reason I didn't want to use snapshot tests is that the errors rely on part on the error produced by SQLite. But probably they are stable enough.

Reply:
I was a bit reluctant on snapshot tests as they rely on the database error as well. But probably they are stable enough.

Reply:
Ooooh, that makes sense. You could maybe use `transform` to rip out the database specific stuff?

Reply:
I simply replaced everything after the last `Caused by error:` with `dummy DBI error`. This should be sufficiently stable.